kratos/solving_strategies/schemes/residual_based_bossak_velocity_scheme.h

```diff
@@ -0,0 +1,805 @@
+/**
+ * It can be used for either first- or second-order time derivatives. Elements
+ * and conditions must provide a specialization of DerivativesExtension via
+ * their data value container, which allows the scheme to operate independently
+ * of the variable arrangements in the element or condition.
+ */
+template <class TSparseSpace, class TDenseSpace>
+class ResidualBasedBossakVelocityScheme : public Scheme<TSparseSpace, TDenseSpace>
+{
+public:
+    ///@name Type Definitions
+    ///@{
+
+    KRATOS_CLASS_POINTER_DEFINITION(ResidualBasedBossakVelocityScheme);
+
+    typedef Scheme<TSparseSpace, TDenseSpace> BaseType;
+
+    typedef typename BaseType::TSystemMatrixType SystemMatrixType;
+
+    typedef typename BaseType::TSystemVectorType SystemVectorType;
+
+    typedef typename BaseType::LocalSystemVectorType LocalSystemVectorType;
+
+    typedef typename BaseType::LocalSystemMatrixType LocalSystemMatrixType;
+
+    typedef typename BaseType::DofsArrayType DofsArrayType;
+
+    typedef std::vector<Dof<double>::Pointer> DofsVectorType;
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor.
+
+    ResidualBasedBossakVelocityScheme(const double AlphaBossak,
+                                      const bool UpdateAcceleration = true,
+                                      const bool UpdateDisplacement = true)
+        : mAlphaBossak(AlphaBossak),
+          mUpdateAcceleration(UpdateAcceleration),
+          mUpdateDisplacement(UpdateDisplacement)
+    {
+        KRATOS_INFO("ResidualBasedBossakVelocityScheme")
+            << " Using bossak velocity scheme with alpha_bossak = " << std::scientific
+            << mAlphaBossak << " [UpdateAcceleration: " << mUpdateAcceleration
+            << ", UpdateDisplacement: " << mUpdateDisplacement << "]\n";
+
+        // Allocate auxiliary memory.
+        int num_threads = OpenMPUtils::GetNumThreads();
+
+        mMassMatrix.resize(num_threads);
+        mDampingMatrix.resize(num_threads);
+        mValuesVector.resize(num_threads);
+        mSecondDerivativeValuesVector.resize(num_threads);
+        mSecondDerivativeValuesVectorOld.resize(num_threads);
+        mIndirectCurrentVelocityVector.resize(num_threads);
+        mIndirectCurrentAccelerationVector.resize(num_threads);
+        mIndirectCurrentDisplacementVector.resize(num_threads);
+        mIndirectOldVelocityVector.resize(num_threads);
+        mIndirectOldAccelerationVector.resize(num_threads);
+        mIndirectOldDisplacementVector.resize(num_threads);
+        mVelocityDofVector.resize(num_threads);
+    }
+
+    /// Destructor.
+    ~ResidualBasedBossakVelocityScheme() override
+    {
+    }
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    void Initialize(ModelPart& rModelPart) override
+    {
+        KRATOS_TRY;
+
+        BaseType::Initialize(rModelPart);
+
+        InitializeNodeNeighbourCount(rModelPart.Nodes());
```

Comment:
This seems like a very specialized operation that doesn't fit in such a generalized class. Also the fact that it is set but never used by this class is odd.

Reply:
removed